packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/point_of_interest.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'types.dart';
+
+/// A pair of latitude and longitude coordinates, stored as degrees.
+@immutable
+class PointOfInterest {
+  /// Creates a PointOfInterest
+  const PointOfInterest(this.position, this.name, this.placeId);
+
+  /// The LatLng of the POI.
+  final LatLng position;
+
+  /// The name of the POI.
+  final String name;
```

Comment:
Retrieving the POI name on the web requires an additional API call to the "Places API", apparently.

Can `name` be made nullable (`String?`), so the web implementation can create a `PointOfInterest` object with extra setup? Or can `name` be a `Future<String?>` so it can be computed asynchronously later?

See `getPlaceInformation(placeId)` in [this example](https://developers.google.com/maps/documentation/javascript/examples/event-poi#maps_event_poi-javascript) 

(More info about the [Places API](https://developers.google.com/maps/documentation/places/web-service))

Reply:
(Web can always create a fully populated PointOfInterest and inject it later into the stream, but it'd require additional setup for users (sign up/authorize the API key), and the code there would always fetch the `name` from another API, regardless of what the programmer wants to do with the POI object)

Reply:
`name` can be nullable. I have pushed the modification to make it nullable. I have made the necessary changes to allow mutability in the PointOfInterest class so that the value of `name` can be set later. By keeping name nullable, we maintain consistency with other event handlers like `onTap` and `onLongPress`, which typically use non-nullable values. Using `Future<String?>` would introduce a different behavior for the `onPoiClicked` event and make it less consistent with the existing event handlers. But to be honest I am just starting with dart maybe there is a better solution someone could come up with.

Reply:
@bndos I think nullable `String?` is good enough. If we wanted to make it a Future, it'd probably complicate logic a lot (because that future would be probably resolved differently on each platform). Thanks!

Reply:
@ditman Thank you for providing your feedback!